pkg/controller/activegate/activegate_update.go

```diff
@@ -41,17 +60,17 @@ func (r *ReconcileActiveGate) updatePods(
 	return nil, nil
 }
 
-func (r *ReconcileActiveGate) findOutdatedPods(
+func (updateService *activeGateUpdateService) FindOutdatedPods(
+	r *ReconcileActiveGate,
 	logger logr.Logger,
-	instance *dynatracev1alpha1.ActiveGate,
-	isLatestFn func(logger logr.Logger, image string, imageID string, imagePullSecret *corev1.Secret) (bool, error)) ([]corev1.Pod, error) {
+	instance *dynatracev1alpha1.ActiveGate) ([]corev1.Pod, error) {
 	pods, err := r.findPods(instance)
 	if err != nil {
 		logger.Error(err, "failed to list pods")
 		return nil, err
 	}
 
-	var outdatedPods []corev1.Pod
+	outdatedPods := make([]corev1.Pod, 0)
```

Comment:
Was this a warning from some tool? My understanding is that the former is [preferred](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices).